amazon/ion/writer_text.py

```diff
@@ -174,9 +174,20 @@ def _bytes_datetime(dt):
 
     fractional_precision = getattr(original_dt, TIMESTAMP_FRACTION_PRECISION_FIELD, MICROSECOND_PRECISION)
     if fractional_precision:
-        fractional = dt.strftime('%f')
-        assert len(fractional) == MICROSECOND_PRECISION
-        if fractional[fractional_precision:] != ('0' * (MICROSECOND_PRECISION - fractional_precision)):
+        if fractional_precision <= MICROSECOND_PRECISION:
```

Comment:
`fractional_precision` should always be used when present. If not present, the existing logic can be used.

Reply:
Could you clarify this? I believe the only way this existing logic can be used is if fractional_precision is present, as seen in line 176?

Reply:
Sorry, I meant `fractional_seconds` should always be used when present. And since we're enforcing at Timestamp construction that it will be present whenever there are fractional seconds, the existing logic can be replaced by logic that writes the `fractional_seconds` in the proper form.

Reply:
Done. Please review.